tests/protocol/test_wire.py

```diff
@@ -362,7 +362,7 @@ def test_send_event(self, mock_http_request, *args):
 
         event_str = u'a test string'
         client = WireProtocol(WIRESERVER_URL).client
-        client.send_event("foo", event_str)
+        client.send_event("foo", event_str.encode('utf-8'))
```

Comment:
Should inputs to send_event be of type str, or of type ustr? This maybe harkens back to my prior comment about keeping encode secluded within communicating with the outside world.

Reply:
I'm assuming that this is, like explained in the other comment, designed to keep changes out of the send_event function.

Reply:
Technically it should be ustr (which is the same as str for Py3) here and only be encoded in the restutil function which actually sends the data out. But since right now I'm only focusing the changes for the telemetry event, we're encoding the string in this function which is just 1 level up from the restutil function to limit the blast radius of this change. 

Does that make sense? We can discuss more on this offline if you have additional doubts.